vulkan_helpers/helper_functions.cpp

```diff
@@ -1144,10 +1144,6 @@ VkSwapchainKHR CreateDefaultSwapchain(
     uint32_t maxSwapchains =
         std::max(surface_caps.maxImageCount, surface_caps.minImageCount + 1);
```

Comment:
This seems wrong. what happens if maxImageCount == miminImageCount, then maxSwapchains will be > maxImageCount

Reply:
I think this should be min. But this is wrong on master just to let you know. I will fix it in my branch. I don't know if any other example would be affected from this.

Reply:
I think what was intended here is clamping the requested image count between min and max capability. I don't know what is the result in master does but I think correct calculation should be: 

```
uint32_t minSwapchains =
        std::max(surface_caps.minImageCount + 1, min_image_count + 1);

    uint32_t numSwapchain =
        std::min(surface_caps.maxImageCount, minSwapchains);
```